### PR TITLE
Fixes regression of functionality that allows for extra volumes and mounts regardless of whether persistence and ssh is enabled

### PR DESCRIPTION
**Pull request by sheroy**
#### What this PR does / why we need it:
Fixes regression of functionality that allows for extra volumes and mounts regardless of whether persistence and ssh is enabled.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #9873

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [ ] Variables are documented in the README.md

### CODE REVIEW

File: stable/gocd/templates/gocd-agent-deployment.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ spec:
         component: agent
     spec:
       serviceAccountName: {{ template "gocd.agentServiceAccountName" . }}
-      {{- if or .Values.agent.persistence.enabled .Values.agent.security.ssh.enabled }}
+      {{- if or .Values.agent.persistence.enabled (or .Values.agent.security.ssh.enabled .Values.agent.persistence.extraVolumes) }}
```
Comment (varshavaradarajan) [inline stable/gocd/templates/gocd-agent-deployment.yaml:30]: @sheroy - thanks, had missed this while checking #9873 . Please check gocd-server-deployment.yaml. You need to add this check for volumeMounts in the container spec as well, I believe.

Comment (sheroy) [reply stable/gocd/templates/gocd-agent-deployment.yaml:30]: @varshavaradarajan, Can you point me to the line# that seems problematic in gocd-server-deployment.yaml please? As far as I can see, the server deployment yaml looks fine. This is just a fix to a regression introduced by another PR.

Comment (varshavaradarajan) [reply stable/gocd/templates/gocd-agent-deployment.yaml:30]: @sheroy - Nevermind. I thought that the extraVolumes change at https://github.com/helm/charts/blame/master/stable/gocd/templates/gocd-agent-deployment.yaml#L116 was missing. Ignore the comment. I'll test this a bit later and merge if that's alright.

Comment (sheroy) [reply stable/gocd/templates/gocd-agent-deployment.yaml:30]: Sure, that's fine. Thanks @varshavaradarajan.

**Pull request merged**